src/components/Dataset/Form/Steps/step-2.jsx

```diff
@@ -30,14 +32,22 @@ class Step2 extends Step {
   */
   getHint() {
     const { form } = this.state;
-    return this.providerDictionary[form.provider].connectorUrlHint;
+    return CONNECTOR_TYPES_DICTIONARY[form.connectorType][form.provider].connectorUrlHint;
   }
 
   render() {
     const hint = this.getHint();
+    const { provider } = this.state.form;
+    const connectorType = this.state.form.connectorType;
```

Comment:
Why don't you get the connectorType as you are getting the provider? const { provider, connectorType }. Try to use always the short es6 way